pkg/operation/botanist/component/extensions/operatingsystemconfig/downloader/templates/scripts/download-cloud-config.tpl.sh

```diff
@@ -10,29 +10,48 @@ TOKEN_SECRET_NAME="{{ .tokenSecretName }}"
 
 PATH_CLOUDCONFIG_DOWNLOADER_CLIENT_CERT="{{ .pathCredentialsClientCert }}"
 PATH_CLOUDCONFIG_DOWNLOADER_CLIENT_KEY="{{ .pathCredentialsClientKey }}"
-PATH_BOOTSTRAP_TOKEN="{{ .pathBootstrapToken }}"
 PATH_CLOUDCONFIG_DOWNLOADER_TOKEN="{{ .pathCredentialsToken }}"
+PATH_BOOTSTRAP_TOKEN="{{ .pathBootstrapToken }}"
+PATH_EXECUTOR_SCRIPT="{{ .pathDownloadedExecutorScript }}"
+PATH_EXECUTOR_SCRIPT_CHECKSUM="{{ .pathDownloadedChecksum }}"
+
+mkdir -p "{{ .pathDownloadsDirectory }}"
 
 function readSecret() {
   wget \
     -qO- \
-    --header         "Accept: application/yaml" \
     --ca-certificate "{{ .pathCredentialsCACert }}" \
     "${@:2}" "$(cat "{{ .pathCredentialsServer }}")/api/v1/namespaces/kube-system/secrets/$1"
 }
 
+function readSecretFull() {
+  readSecret "$1" "--header=Accept: application/yaml" "${@:2}"
+}
+
+function readSecretMeta() {
+  readSecret "$1" "--header=Accept: application/yaml;as=PartialObjectMetadata;g=meta.k8s.io;v=v1,application/yaml;as=PartialObjectMetadata;g=meta.k8s.io;v=v1" "${@:2}"
```

Comment:
I'm wondering why is the value of the accept header duplicated, I couldn't spot the difference.

```
--header=Accept:

application/yaml;as=PartialObjectMetadata;g=meta.k8s.io;v=v1,
application/yaml;as=PartialObjectMetadata;g=meta.k8s.io;v=v1
```

Note that this approach reduces the network bandwidth between the client and the api server; but it (very likely) doesn't make a difference between the api server and etcd: for that I think we would need a list+watch approach.
Regarding the api server - etcd traffic, we'll have to wait for better monitoring so that we can optimize the low hanging fruits _first_, and this (might) turn out to be a low hanging fruit due to the frequency and execution on each node - let's see.